pkg/cfn/builder/iam.go

```diff
@@ -224,10 +224,27 @@ func (rs *IAMServiceAccountResourceSet) AddAllResources() error {
 		PermissionsBoundary:      rs.spec.PermissionsBoundary,
 		RoleName:                 rs.spec.RoleName,
 	}
-	role.ManagedPolicyArns = append(role.ManagedPolicyArns, rs.spec.AttachPolicyARNs...)
+	for _, arn := range rs.spec.AttachPolicyARNs {
+		role.ManagedPolicyArns = append(role.ManagedPolicyArns, arn)
+	}
+
+	wellKnownPolicies := createWellKnownPolicies(rs.spec.WellKnownPolicies)
+
+	for _, p := range wellKnownPolicies {
+		for _, np := range p.NamedPolicies {
+			role.ManagedPolicyArns = append(role.ManagedPolicyArns, makePolicyARN(np))
+		}
+	}
 
 	roleRef := rs.template.NewResource("Role1", role)
 
+	for _, p := range wellKnownPolicies {
+		if p.Name != "" {
+			doc := cft.MakePolicyDocument(p.Statements...)
+			rs.template.AttachPolicy(p.Name, roleRef, doc)
+		}
```

Comment:
Is it possible to move this up under line 236 in the other loop over `wellKnownPolicies`? Seems a shame to do it twice if we can get away with not.

Reply:
i've solved it by just returning a tuple of lists instead of a list of a sum type imitation.

Reply:
looks good 👍